learning/prompts/documentation-lookup/08_basic_windowing.md

```diff
@@ -0,0 +1,29 @@
+Prompt:
+What is Windowing in Apache Beam?
+Response:
+Windowing is a key concept in stream processing, as it allows you to divide data streams into logical units for efficient and correct parallel processing.
+
+With an unbounded data set, collecting all of the elements is impossible since new elements are constantly being added. In the Beam model, any PCollection (including unbounded PCollections) can be subdivided into [logical windows](https://beam.apache.org/documentation/programming-guide/#windowing-basics). Grouping transforms then consider each PCollection’s elements on a per-window basis.
+
+Since Beam's default windowing strategy is to assign each element to a single, global window, you must explicitly specify a [windowing function](https://beam.apache.org/documentation/programming-guide/#setting-your-pcollections-windowing-function) for your pipeline.
```

Comment:
```suggestion
Because Beam's default windowing strategy is to assign each element to a single, global window, you must explicitly specify a [windowing function](https://beam.apache.org/documentation/programming-guide/#setting-your-pcollections-windowing-function) for your pipeline.
```